subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -822,8 +825,14 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 		}
 	}
 
+	if (client_ctx->set_socketoptions) {
```

Comment:
The function docs says: `Client can override default socket options`.

Wouldn't it then make sense to call this callback **after** the default options are set (i. e. just before connect)?

Reply:
I specificly wanted to completely skip the default option if custom ones are provided.
Same pattern what the `load_credentials()` callback uses.

This gives option for some exotic IP stacks don't even support our default, someone may the completely skip those and provide their own.

Reply:
Ok then